client/components/command-palette/index.tsx

```diff
@@ -200,6 +201,42 @@ function CommandInput( {
 	);
 }
 
+interface NotFoundMessageProps {
+	selectedCommandName: string;
+	search: string;
+	emptyListNotice?: string;
+	currentRoute: string | null;
+}
+
+const NotFoundMessage = ( {
+	selectedCommandName,
+	search,
+	emptyListNotice,
+	currentRoute,
+}: NotFoundMessageProps ) => {
+	const dispatch = useDispatch();
+	const trackNotFoundDebounced = useDebounce( () => {
+		dispatch(
+			recordTracksEvent( 'calypso_hosting_command_palette_not_found', {
+				command: selectedCommandName,
+				current_route: currentRoute,
+				search_text: search,
+			} )
+		);
+	}, 600 );
+
+	useEffect( () => {
+		// Track search queries for root and nested commands
+		// Track default zero results for nested commands
+		if ( search || ( ! search && selectedCommandName ) ) {
```

Comment:
I don't think the data is that useful for `search && selectedCommandName`, i.e. I don't really care what sites are missing in their search.

What do you think about only tracking for "search queries for root commands" and "zero results for nested commands"?

Reply:
Agreed! Updated here https://github.com/Automattic/wp-calypso/pull/85647/commits/840dc875dcc52ffe150ced86524262379cb57b0a and https://github.com/Automattic/wp-calypso/pull/85647/commits/b14fc6b64cb1f88ebd5c1c785bb6bb135349aed8